packages/shared/pkg/logger/grpc.go

```diff
@@ -24,23 +45,28 @@ func GRPCLogger(l *zap.Logger) logging.Logger {
 		}
 
 		for i := 0; i < len(fields)-1; i += 2 {
-			key := fields[i]
+			key := fields[i].(string)
```

Comment:
To prevent edge case panic catastrophe, we should check for success:


```suggestion
			key, ok := fields[i].(string)
			if !ok {
				l.Warn("field is not a string", zap.Any(fields[i]))
				continue
			}
```